### DISCUSSION TITLE
Are dependency files (.d) supposed to be hard linked with `CCACHE_HARDLINK=1` option?

### DISCUSSION THREAD

**Discussion by ericch1** [ccache/ccache#1209] [Questions & answers] [answered]
Hi,

just noticed that when I use ccache with `CCACHE_HARDLINK=1` option, the .o file is a hard link, but not the .d file.

Do I miss something in my configuration or is it a normal behavior?

Here is my simple test:
```
echo  "void foo(){}" > foo.cc
export CCACHE_HARDLINK=1
export CCACHE_SLOPPINESS=include_file_mtime
export CCACHE_PREFIX=/opt/icecream-1.2/bin/icecc
export CCACHE_COMMENTS=1
/usr/lib64/ccache/g++ -MD -c foo.cc 
ls -la foo*
-rw------- 1 eric giref   13 Nov  3 05:17 foo.cc
-rw------- 1 eric giref   41 Nov  3 05:24 foo.d
-r--r--r-- 2 eric giref 1208 Nov  3 05:24 foo.o
```

as you can see, `foo.o` as a inode ref count of 2 but not `foo.d`.

Thanks,

Eric

**Comment by jrosdahl** [comment, marked as answer]
It's intentional and expected. [The documentation of hard_link](https://ccache.dev/manual/4.7.2.html#config_hard_link) says _"If true, ccache will attempt to use hard links to store and fetch cached **object files**"_.

The reason is described here: https://github.com/ccache/ccache/blob/157a18f250b31a4b280b994eb277e7ce7d569d4a/src/core/Result.cpp#L88-L103

**Reply by ericch1** [reply]
Ok, that's clear now! Thanks for the detailed answer! :)
